File: pkg/dev_compiler/test/codegen_expected/lib/typed_data/float32x4_unbox_phi_test.js
dart_library.library('lib/typed_data/float32x4_unbox_phi_test', null, /* Imports */[
  'dart_sdk',
  'expect'
], function load__float32x4_unbox_phi_test(exports, dart_sdk, expect) {
  'use strict';
  const core = dart_sdk.core;
  const typed_data = dart_sdk.typed_data;
  const dart = dart_sdk.dart;
  const dartx = dart_sdk.dartx;
  const expect$ = expect.expect;
  const float32x4_unbox_phi_test = Object.create(null);
  let Float32x4ListTodouble = () => (Float32x4ListTodouble = dart.constFn(dart.definiteFunctionType(core.double, [typed_data.Float32x4List])))();
  let VoidTodynamic = () => (VoidTodynamic = dart.constFn(dart.definiteFunctionType(dart.dynamic, [])))();
  float32x4_unbox_phi_test.testUnboxPhi = function(data) {
    let res = typed_data.Float32x4.zero();
    for (let i = 0; i < dart.notNull(data.length); i++) {
      res = res['+'](data.get(i));
    }
    return dart.notNull(res.x) + dart.notNull(res.y) + dart.notNull(res.z) + dart.notNull(res.w);
  };
  dart.fn(float32x4_unbox_phi_test.testUnboxPhi, Float32x4ListTodouble());
  float32x4_unbox_phi_test.main = function() {
    let list = typed_data.Float32x4List.new(10);
    let floatList = typed_data.Float32List.view(list.buffer);
    for (let i = 0; i < dart.notNull(floatList[dartx.length]); i++) {
      floatList[dartx.set](i, i[dartx.toDouble]());
    }
    for (let i = 0; i < 20; i++) {
      let r = float32x4_unbox_phi_test.testUnboxPhi(list);
      expect$.Expect.equals(780.0, r);
    }
  };
  dart.fn(float32x4_unbox_phi_test.main, VoidTodynamic());
  // Exports:
  exports.float32x4_unbox_phi_test = float32x4_unbox_phi_test;
});

File: tests/lib_2/typed_data/float32x4_unbox_phi_test.dart
// Copyright (c) 2013, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// VMOptions=--max_deoptimization_counter_threshold=1000 --optimization-counter-threshold=10 --no-background-compilation

// Library tag to be able to run in html test framework.
library float32x4_unbox_regress_test;

import 'dart:typed_data';
import 'package:expect/expect.dart';

double testUnboxPhi(Float32x4List data) {
  var res = new Float32x4.zero();
  for (int i = 0; i < data.length; i++) {
    res += data[i];
  }
  return res.x + res.y + res.z + res.w;
}

main() {
  Float32x4List list = new Float32x4List(10);
  Float32List floatList = new Float32List.view(list.buffer);
  for (int i = 0; i < floatList.length; i++) {
    floatList[i] = i.toDouble();
  }
  for (int i = 0; i < 20; i++) {
    double r = testUnboxPhi(list);
    Expect.equals(780.0, r);
  }
}
